libcni/api.go

```diff
@@ -165,33 +169,41 @@ func (c *CNIConfig) ensureExec() invoke.Exec {
 	return c.exec
 }
 
-func getResultCacheFilePath(netName string, rt *RuntimeConf) string {
+type cacheFileType string
+
+const (
+	cacheFileTypeResult cacheFileType = "results"
+	cacheFileTypeConfig cacheFileType = "config"
+)
+
+func getCacheFilePath(fileType cacheFileType, netName string, rt *RuntimeConf) string {
 	cacheDir := rt.CacheDir
 	if cacheDir == "" {
 		cacheDir = CacheDir
 	}
-	return filepath.Join(cacheDir, "results", fmt.Sprintf("%s-%s-%s", netName, rt.ContainerID, rt.IfName))
+	fName := fmt.Sprintf("%s-%s-%s", netName, rt.ContainerID, rt.IfName)
+	return filepath.Join(cacheDir, string(fileType), base64.StdEncoding.EncodeToString([]byte(fName)))
 }
 
 func setCachedResult(result types.Result, netName string, rt *RuntimeConf) error {
 	data, err := json.Marshal(result)
 	if err != nil {
 		return err
 	}
-	fname := getResultCacheFilePath(netName, rt)
+	fname := getCacheFilePath(cacheFileTypeResult, netName, rt)
 	if err := os.MkdirAll(filepath.Dir(fname), 0700); err != nil {
 		return err
 	}
 	return ioutil.WriteFile(fname, data, 0600)
 }
 
 func delCachedResult(netName string, rt *RuntimeConf) error {
-	fname := getResultCacheFilePath(netName, rt)
+	fname := getCacheFilePath(cacheFileTypeResult, netName, rt)
 	return os.Remove(fname)
 }
 
 func getCachedResult(netName, cniVersion string, rt *RuntimeConf) (types.Result, error) {
-	fname := getResultCacheFilePath(netName, rt)
+	fname := getCacheFilePath(cacheFileTypeResult, netName, rt)
```

Comment:
For backwards compat with older code, could we fall back to the un-base64-ed name here? Otherwise things may not work as well when a project revendors CNI and had already been using the previous code.